### PR TITLE
[BEAM-2281] Refactoring literal expression type conversion 

### PR DESCRIPTION
**Pull request by vectorijk**
call `SqlFunctions` in `BeamSqlFnExecutor` for refactoring literal expression type conversion.

------------------------

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

It will help us expedite review of your Pull Request if you tag someone (e.g. `@username`) to look at it.

### CODE REVIEW

**Comment by vectorijk** [comment]
R: @kennknowles 
cc: @apilloud @akedin @XuMingmin

**Comment by vectorijk** [comment]
retest this please

File: sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/BeamSqlFnExecutor.java [hunk 1]
```diff
@@ -171,36 +171,39 @@ static BeamSqlExpression buildExpression(RexNode rexNode) {
...
               break;
             default:
               throw new IllegalStateException(
-                  "type/realType mismatch: " + type + " VS " + realType);
-          }
-        } else if (type == SqlTypeName.DOUBLE) {
-          Double rawValue = (Double) value;
-          if (realType == SqlTypeName.FLOAT) {
-            realValue = rawValue.floatValue();
+                  "type / realType mismatch: " + type + " <-> " + realType);
```
Comment (akedin) [inline sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/BeamSqlFnExecutor.java]: nit: explain a bit more in the message? E.g. "Attempted convert node zzz of type xxx to yyy. This conversion is unsupported"

Comment (vectorijk) [reply sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/BeamSqlFnExecutor.java]: Addressed. Update later.

**Comment by vectorijk** [comment]
pushed the update. wait jenkin turns green and happy.

**Comment by vectorijk** [comment]
run java precommit

**Pull request merged**